### PR TITLE
fix: avoid trailing ? when no query string present

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Avoid trailing ? when no query string present

## Motivation / Context
Fix edge case

## Testing Instructions / How This Has Been Tested
Tests should pass. Blocked by #54

### CODE REVIEW

File: src/Http/Client.php [hunk 1]
```diff
@@ -113,7 +113,7 @@ public function request(
         if (!empty($query_string)) {
             $full_query_string = !empty($full_query_string) ? $full_query_string . '&' . $query_string : $query_string;
         }
-        $url = $endpoint . '?' . $full_query_string;
+        $url = !empty($full_query_string) ? $endpoint . '?' . $full_query_string : $endpoint;
```
Comment (apotek) [inline src/Http/Client.php]: This works well, but I think either of the below might be more readable.

```php
// Either
$url = $endpoint;
if ($full_query_string !== '') {
  $url =. '?' . $full_query_string;
}

// Or
if ($full_query_string !== '') {
  $url = $endpoint . '?' . $full_query_string;
}
else {
  $url = $endpoint;
}
```

File: tests/Http/ClientTest.php [hunk 2]
```diff
@@ -53,4 +57,21 @@ public function testSetOauthToken(): void
         $this->assertTrue($client->oauth2);
         $this->assertSame('example-token', $client->token);
     }
+
```
Comment (apotek) [inline tests/Http/ClientTest.php:60]: Thanks for the test on this! 💯 🥇

**Review by apotek — APPROVED** [review]
Thanks for the fix and the test!

**Review by apotek — APPROVED** [review]
Perfect. Thank you Mark.
